src/aas-api-webapp-repository/Controllers/AasRepositoryApi.cs

```diff
@@ -0,0 +1,95 @@
+        {
+            _configuration = config ??
+                             throw new ArgumentNullException(nameof(config));
+            _mapper = mapper ??
+                      throw new ArgumentNullException(nameof(mapper));
+
+            repository = new AASRepositoryFactory(adtInteractions,mapper).CreateAASRepositoryForADT(config["ADT_SERVICE_URL"]) ??
+                         throw new ArgumentNullException(); ;
+        }
+
+
+        /// <summary>
+        /// Returns all Asset Administration Shells
+        /// </summary>
+        /// <param name="assetIds">The key-value-pair of an Asset identifier</param>
+        /// <param name="idShort">The Asset Administration Shell’s IdShort</param>
+        /// <response code="200">Requested Asset Administration Shells</response>
+        [HttpGet]
+        [ValidateModelState]
+        [SwaggerOperation("GetAllAssetAdministrationShells")]
+        [SwaggerResponse(statusCode: 200, type: typeof(List<AssetAdministrationShell>), description: "Requested Asset Administration Shells")]
+        public async Task<ActionResult<List<AssetAdministrationShell>>> GetAllAssetAdministrationShells([FromQuery(Name = "assetIds")] List<IdentifierKeyValuePair> assetIds, [FromQuery] string idShort)
+        {
+            //TODO: Uncomment the next line to return response 200 or use other options such as return this.NotFound(), return this.BadRequest(..), ...
+
+            var aas = await repository.GetAllAdministrationShells();
+            return Ok(aas);
+
+            //if (idShort != null && idShort.Length > 0)
+            //{
+            //    return new ObjectResult(repository.GetAllAssetAdministrationShellsByIdShort(idShort).GetAwaiter().GetResult());
+            //}
+            //else
+            //{
+            //    if (assetIds != null && assetIds.Count > 0)
+            //        return new ObjectResult(repository.GetAllAssetAdministrationShellsByAssetId(assetIds).GetAwaiter().GetResult());
+            //    else
+            //    {
+            //        var aas = await repository.GetAllAdministrationShells();
+            //        return Ok(aas); //new ObjectResult(.GetAwaiter().GetResult());
+            //    }
+            // }
+        }
+
+        /// <summary>
+        /// Returns a specific Asset Administration Shell
+        /// </summary>
+        /// <param name="aasIdentifier">The Asset Administration Shell’s unique id (BASE64-URL-encoded)</param>
+        /// <response code="200">Requested Asset Administration Shell</response>
+        [HttpGet]
+        [Route("api/v1/shells/{aasIdentifier}")]
+        [ValidateModelState]
+        [SwaggerOperation("GetAssetAdministrationShellById")]
+        [SwaggerResponse(statusCode: 200, type: typeof(AssetAdministrationShell), description: "Requested Asset Administration Shell")]
+        public async Task<ActionResult<AssetAdministrationShell>> GetAssetAdministrationShellById([FromRoute][Required] string aasIdentifier)
```

Comment:
No exception handling? No standard error code handling according to the Standard?

Reply:
Not implemented yet